From dd25e35e0850e011f94748cb061c63b84f7ada2b Mon Sep 17 00:00:00 2001
From: Wyon Bi <bivvy.bi@rock-chips.com>
Date: Fri, 18 Aug 2017 14:27:39 +0800
Subject: [PATCH] drm: bridge: analogix: Avoid timeout warnings in
 analogix_dp_config_video()

[   23.527279] rockchip-dp ff970000.edp: Timeout of video streamclk ok
[   23.527325] rockchip-dp ff970000.edp: unable to config video

Change-Id: Ie2239c2dd773e4cb41f86b46b98367a07a751ff1
Signed-off-by: Wyon Bi <bivvy.bi@rock-chips.com>
---
 drivers/gpu/drm/bridge/analogix/analogix_dp_core.c | 13 ++++---------
 1 file changed, 4 insertions(+), 9 deletions(-)

diff --git a/drivers/gpu/drm/bridge/analogix/analogix_dp_core.c b/drivers/gpu/drm/bridge/analogix/analogix_dp_core.c
index f8495cdeb8da..957d1755250a 100644
--- a/drivers/gpu/drm/bridge/analogix/analogix_dp_core.c
+++ b/drivers/gpu/drm/bridge/analogix/analogix_dp_core.c
@@ -765,7 +765,6 @@ static int analogix_dp_set_link_train(struct analogix_dp_device *dp,
 
 static int analogix_dp_config_video(struct analogix_dp_device *dp)
 {
-	int retval = 0;
 	int timeout_loop = 0;
 	int done_count = 0;
 
@@ -802,8 +801,10 @@ static int analogix_dp_config_video(struct analogix_dp_device *dp)
 	/* Configure video slave mode */
 	analogix_dp_enable_video_master(dp, 0);
 
-	timeout_loop = 0;
+	/* Enable video input */
+	analogix_dp_start_video(dp);
 
+	timeout_loop = 0;
 	for (;;) {
 		timeout_loop++;
 		if (analogix_dp_is_video_stream_on(dp) == 0) {
@@ -821,10 +822,7 @@ static int analogix_dp_config_video(struct analogix_dp_device *dp)
 		usleep_range(1000, 1001);
 	}
 
-	if (retval != 0)
-		dev_err(dp->dev, "Video stream is not detected!\n");
-
-	return retval;
+	return 0;
 }
 
 static void analogix_dp_enable_scramble(struct analogix_dp_device *dp,
@@ -925,9 +923,6 @@ static void analogix_dp_commit(struct analogix_dp_device *dp)
 		if (drm_panel_enable(dp->plat_data->panel))
 			DRM_ERROR("failed to enable the panel\n");
 	}
-
-	/* Enable video */
-	analogix_dp_start_video(dp);
 }
 
 int analogix_dp_get_modes(struct drm_connector *connector)
-- 
2.35.3

